lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -71,10 +71,16 @@ def serializable_hash_for_single_resource(options)
         end
 
         def resource_identifier_type_for(serializer)
-          if ActiveModel::Serializer.config.jsonapi_resource_type == :singular
-            serializer.object.class.model_name.singular
+          if serializer.class._type.nil?
+            if ActiveModel::Serializer.config.type_source == :model
+              type = serializer.object.class.name.demodulize.underscore
+            else
+              type = serializer.class.name.demodulize.chomp('Serializer').underscore
```

Comment:
Also, I'm totally in favor of removing the namespace, but I think it should be done in a separate PR.